modules/open8BidAdapter.js

```diff
@@ -0,0 +1,182 @@
+    var requests = [];
+    for (var i = 0; i < validBidRequests.length; i++) {
+      var bid = validBidRequests[i];
+      var queryString = '';
+      var slotKey = utils.getBidIdParameter('slotKey', bid.params);
+      queryString = utils.tryAppendQueryString(queryString, 'slot_key', slotKey);
+      queryString = utils.tryAppendQueryString(queryString, 'imp_id', generateImpId());
+      queryString = utils.tryAppendQueryString(queryString, 'bid_id', bid.bidId);
+
+      requests.push({
+        method: 'GET',
+        url: URL,
+        data: queryString
+      });
+    }
+    return requests;
+  },
+
+  interpretResponse: function(serverResponse, request) {
+    var bidderResponse = serverResponse.body;
+
+    if (!bidderResponse.isAdReturn) {
+      return [];
+    }
+
+    var ad = bidderResponse.ad;
+
+    const bid = {
+      slotKey: bidderResponse.slotKey,
+      userId: bidderResponse.userId,
+      impId: bidderResponse.impId,
+      media: bidderResponse.media,
+      ds: ad.ds,
+      spd: ad.spd,
+      fa: ad.fa,
+      pr: ad.pr,
+      mr: ad.mr,
+      nurl: ad.nurl,
+      requestId: ad.bidId,
+      cpm: ad.price,
+      creativeId: ad.creativeId,
+      dealId: ad.dealId,
+      currency: ad.currency || 'JPY',
+      netRevenue: true,
+      ttl: 360, // 6 minutes
+    }
+
+    if (ad.adType === AD_TYPE.VIDEO) {
+      const videoAd = bidderResponse.ad.video;
+      Object.assign(bid, {
+        vastXml: videoAd.vastXml,
+        width: videoAd.w,
+        height: videoAd.h,
+        renderer: newRenderer(bidderResponse),
+        adResponse: bidderResponse,
+        mediaType: VIDEO
+      });
+    } else if (ad.adType === AD_TYPE.BANNER) {
+      const bannerAd = bidderResponse.ad.banner;
+      Object.assign(bid, {
+        width: bannerAd.w,
+        height: bannerAd.h,
+        ad: bannerAd.adm,
+        mediaType: BANNER
+      });
+      try {
+        bannerAd.imps.forEach(impTrackUrl => {
+          const tracker = utils.createTrackPixelHtml(impTrackUrl);
+          bid.ad += tracker;
+        });
+      } catch (error) {
+        utils.logError('Error appending imp tracking pixel', error);
+      }
+    }
+    return [bid];
+  },
+
+  getUserSyncs: function(syncOptions, serverResponses) {
+    const syncs = [];
+    if (syncOptions.iframeEnabled && serverResponses.length) {
+      const syncIFs = serverResponses[0].body.syncIFs;
+      if (syncIFs) {
+        syncIFs.forEach(sync => {
+          syncs.push({
+            type: 'iframe',
+            url: sync
+          });
+        });
+      }
+    }
+    if (syncOptions.pixelEnabled && serverResponses.length) {
+      const syncPixs = serverResponses[0].body.syncPixels;
+      if (syncPixs) {
+        syncPixs.forEach(sync => {
+          syncs.push({
+            type: 'image',
+            url: sync
+          });
+        });
+      }
+    }
+    return syncs;
+  },
+  onBidWon: function(bid) {
+    const winUrl = bid.nurl.replace(
```

Comment:
Could add a check here to verify the `nurl` property is populated in the `bid` object.